erpnext/crm/report/opportunity_summary_by_sales_stage/opportunity_summary_by_sales_stage.py

```diff
@@ -0,0 +1,274 @@
+
+	def run(self):
+		self.get_columns()
+		self.get_data()
+		self.get_chart_data()
+
+		return self.columns,self.data,None,self.chart
+
+	def get_columns(self):
+		self.columns = []
+
+		if self.filters.get('based_on') == "Opportunity Owner":
+			self.columns.append({
+				'label': _('Opportunity Owner'),
+				'fieldname': 'opportunity_owner',
+				'width':200
+			})
+		if self.filters.get('based_on') == "Source":
+			self.columns.append({
+				'label': _('Source'),
+				'fieldname': 'source',
+				'fieldtype': 'Link',
+				'options':'Lead Source',
+				'width': 200
+			})
+		if self.filters.get('based_on') == "Opportunity Type":
+			self.columns.append({
+				'label': _('Opportunity Type'),
+				'fieldname': 'opportunity_type',
+				'width': 200
+			})
+
+		
+		self.sales_stage_list = frappe.db.get_list("Sales Stage",pluck="name")
+		for sales_stage in self.sales_stage_list:
+			self.columns.append({
+				'label': _(sales_stage),
+				'fieldname': sales_stage,
+				'width':150
+			})
+
+		return self.columns
+
+	def get_data(self):
+		self.data = []
+		if self.filters.get('based_on') == "Opportunity Owner":
+			sql = "_assign"
+			self.get_data_query(sql)
+		if self.filters.get('based_on') == "Source":
+			sql = "source"
+			self.get_data_query(sql)
+		if self.filters.get('based_on') == "Opportunity Type":
+			sql = "opportunity_type"
+			self.get_data_query(sql)
+
+		self.get_rows()
+	
+	def get_data_query(self,sql):
+		filter_data = self.filters.get('status')
+		
+		if self.filters.get("data_based_on") == "Number":
+			if self.filters.get('status'):
+				self.filters.update({'status':tuple(filter_data)})
+			self.query_result = frappe.db.sql("""select sales_stage,count(name) as count,{sql} from tabOpportunity
+			where {conditions} 
+			group by sales_stage,{sql}""".format(conditions=self.get_conditions(),sql=sql),self.filters,as_dict=1)
+
+		if self.filters.get("data_based_on") == "Amount":
+			if self.filters.get('status'):
+				self.filters.update({'status':tuple(filter_data)})
+			self.query_result = frappe.db.sql("""select sales_stage,sum(opportunity_amount) as amount,{sql} from tabOpportunity
+			where {conditions} 
+			group by sales_stage,{sql}""".format(conditions=self.get_conditions(),sql=sql),self.filters,as_dict=1)
+		
+	def get_rows(self):
+		self.data = []
+		self.get_formatted_data()
+		currency_symbol = self.get_currency()
+		
+		for based_on,data in iteritems(self.formatted_data):
+			if self.filters.get("based_on") == "Opportunity Owner":
+				row = {'opportunity_owner': based_on}
+
+				if self.filters.get("data_based_on") == "Number":
+					for d in self.query_result:
+						sales_stage = d.get('sales_stage')
+						count = data.get(sales_stage)
+						row[sales_stage] = count
+
+				if self.filters.get("data_based_on") == "Amount":
+					for d in self.query_result:
+						sales_stage = d.get('sales_stage')
+						amount = data.get(sales_stage)
+						if amount:
+							row[sales_stage] = str(amount) + currency_symbol
```

Comment:
Amount columns should be of the field type Currency, so you don't need to append the currency symbol like this. Moreover, you will have to convert the amount into company currency. 

For eg: If company currency is INR and there is an opportunity with currency USD, you will have to convert the opportunity amount to INR based on Exchange Rate.